projects/extension/src/background/AppMediator.ts

```diff
@@ -98,6 +101,20 @@ export class AppMediator extends (EventEmitter as { new(): StateEmitter }) {
     return this.#appName;
   }
 
+  /** healthStatus returns the latest health status
+   * of app as set from the callback
+   */
+  get healthStatus(): HealthResponse {
+    return this.#healthStatus as HealthResponse;
+  }
+
+  /** 
+  * returns the healthChecker that the app is using
+  */
+  get healthChecker(): HealthChecker {
+    return this.#healthChecker as HealthChecker;
+  }
```

Comment:
This exposes an implementation detail. Is it really necessary?

Reply:
No it is not. Remained

Reply:
The tests are whatever you want them to be

Reply:
I agree that this should not be exposed. This means that If this is removed - relevant tests should be removed as well as they will not work without it.

Reply:
As agreed this will be removed in order to not expose impl details and along will be removed the relevant tests